execution/aggregate/khashaggregate.go

```diff
@@ -91,12 +91,16 @@ func (a *kAggregate) Next(ctx context.Context) ([]model.StepVector, error) {
 	if err != nil {
 		return nil, err
 	}
-	for i := range a.params {
-		a.params[i] = math.NaN()
-		if i < len(args) {
-			a.params[i] = args[i].Samples[0]
-			a.paramOp.GetPool().PutStepVector(args[i])
+	for i := range args {
```

Comment:
I wonder: why do we need to iterate args for all steps, shouldnt it be constant?

Reply:
IIRC, the param could change if we use something like `topk(scalar(some vector), xxx)`. I don't remember the details so I might be wrong.